jax_galsim/wcs.py

```diff
@@ -0,0 +1,922 @@
+import jax.numpy as jnp
+
+from jax_galsim.gsobject import GSObject
+from jax_galsim.position import Position, PositionI, PositionD
+from jax_galsim.transform import _Transform
+
+import galsim as _galsim
+from jax._src.numpy.util import _wraps
+from jax.tree_util import register_pytree_node_class
+
+
+# We inherit from the reference BaseWCS and only redefine the methods that
+# make references to jax_galsim objects.
+class BaseWCS(_galsim.BaseWCS):
+    @_wraps(_galsim.BaseWCS.toWorld)
+    def toWorld(self, *args, **kwargs):
+        if len(args) == 1:
+            if isinstance(args[0], GSObject):
+                return self.profileToWorld(*args, **kwargs)
+            else:
+                return self.posToWorld(*args, **kwargs)
+        elif len(args) == 2:
+            if self.isCelestial():
+                return self.xyToradec(*args, **kwargs)
+            else:
+                return self.xyTouv(*args, **kwargs)
+        else:
+            raise TypeError("toWorld() takes either 1 or 2 positional arguments")
+
+    @_wraps(_galsim.BaseWCS.posToWorld)
+    def posToWorld(self, image_pos, color=None, **kwargs):
+        if color is None:
+            color = self._color
+        if not isinstance(image_pos, Position):
+            raise TypeError("image_pos must be a PositionD or PositionI argument")
+        return self._posToWorld(image_pos, color=color, **kwargs)
+
+    @_wraps(_galsim.BaseWCS.toImage)
+    def toImage(self, *args, **kwargs):
+        if len(args) == 1:
+            if isinstance(args[0], GSObject):
```

Comment:
One of the options in galsim here is using a `Shear` object but we don't have one yet so its not included as one of the `if` statements here. Should we add a `#TODO` so we don't forget to add this or will the tests catch it?